### PR TITLE
Guardian Labs: New `LabsSection` front container for paid content 

### PR DESCRIPTION
**Pull request by alinaboghiu**
## What does this change?

Closes https://github.com/guardian/dotcom-rendering/issues/5150

Introduces a new LabsSection component in DCR to render what is marked as 'Branded' content in the Fronts tool. 

## Why?

Sadly neither Section nor FrontSection support the use case for the Guardian Labs container. In both cases it's impossible to colour the background of the left column all the way to the edges (because container padding in Section and because of the the css grid gap in FrontSection). Hence the new component. Labs is special. 

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |

[before]: https://example.com/before.png
[after]: https://example.com/after.png

### CODE REVIEW

File: dotcom-rendering/src/model/decideContainerPalette.ts [hunk 1]
```diff
@@ -20,6 +21,6 @@ export const decideContainerPalette = (
 	if (palettes?.includes('BreakingPalette')) return 'BreakingPalette';
 	if (palettes?.includes('SpecialReportAltPalette'))
 		return 'SpecialReportAltPalette';
-	if (palettes?.includes('Branded')) return 'Branded';
+	if (palettes?.includes('Branded') && allCardsHaveBranding) return 'Branded';
```
Comment (alinaboghiu) [inline dotcom-rendering/src/model/decideContainerPalette.ts:24]: Even if a container has the 'Branded' label in the fronts tool, it will behave like a regular container if any of the cards are for actual (unpaid) Guardian content, effectively removing the Branded palette

File: dotcom-rendering/src/types/palette.ts [hunk 2]
```diff
@@ -152,29 +152,33 @@ export type Palette = {
 
 export type ContainerOverrides = {
 	text?: {
```
Comment (alinaboghiu) [inline dotcom-rendering/src/types/palette.ts:154]: **tl;dr** I have made all these properties optional because the Guardian Labs container does not override most of these.

I think this is ok, as different containers types should be able to override different parts of the palette. I think it's less weird than providing a 'Branded' override colour for each of these, when if fact it would be the same as the default value, and if the default value would change we would want the Labs container to change too.  

Keen to hear more thoughts on this tho because it's not ideal. An option would be for example convincing the Labs team to have their entire background team, not just the left column. But this might look weird with the teal card headlines. So a more comprehensive redesign would be needed to bring this container in line with the rest.

File: dotcom-rendering/src/web/lib/decideContainerOverrides.ts [hunk 3]
```diff
@@ -170,7 +207,7 @@ const textDynamoKicker = (
 };
 
 const textDynamoSublinkKicker = (
```
Comment (alinaboghiu) [inline dotcom-rendering/src/web/lib/decideContainerOverrides.ts:209]: We were never using this override for Labs

File: dotcom-rendering/src/web/layouts/FrontLayout.tsx [hunk 4]
```diff
@@ -346,6 +347,49 @@ export const FrontLayout = ({ front, NAV }: Props) => {
 						);
 					}
 
+					if (
+						collection.containerPalette === 'Branded' &&
+						renderAds
```
Comment (alinaboghiu) [inline dotcom-rendering/src/web/layouts/FrontLayout.tsx:352]: we still render labs container even with reject all - double check

Comment (arelra) [reply dotcom-rendering/src/web/layouts/FrontLayout.tsx:352]: I thought it might be useful to expand on the current state of ad free and `canRenderAds`.

Thanks to our ad free expert @emma-imber we have in depth documentation here:
https://docs.google.com/document/d/10Mxxta6ypxAduVXJFp19uRoWtALQmAubdKtBwq7AqZ0/edit#heading=h.8g8uwordyell

TLDR:

`isAdFree` is available server side via a Fastly header but originates as a client side cookie, set in the following scenarios:

- the user rejects all or does not give consent to googletag
- the user is a subscriber

On DCR the `canRenderAds` function checks `isAdFree` and `shouldHideAds` (e.g. sensitive article content).

So wrt this PR if the user 'rejects all' I expect the labs container will _not_ be shown.

However since the recent introduction of consentless ads we are planning to decouple 'reject all' from 'isAdFree'.

So in future we will move to a model where 'reject all' users will see a consentless ad experience which includes labs containers. There is no immediate impact to any current labs work as we will handle the migration but just as an fyi.

**Review by arelra — APPROVED** [review]
Nice work 💎 ! Thanks for the walk-through

File: dotcom-rendering/src/web/layouts/FrontLayout.tsx (hunk 4 above)
Comment (emma-imber) [reply dotcom-rendering/src/web/layouts/FrontLayout.tsx:352]: A point that just occurred to me that I thought I should add - at the moment, with the consentless advertising switch on, we unset GU_AF1 if the reason for it was reject all. I suspect at the moment because of this, the labs container will show in a reject all environment. But if we turn the consentless switch back off, reject all would then hide the labs container. I think whether or not the labs container shows might currently be coupled to the consentless advertising switch?

Comment (arelra) [reply dotcom-rendering/src/web/layouts/FrontLayout.tsx:352]: Ah good point @emma-imber ! I think in the new consentless world we want the following:

- the user rejects all: Labs should render
- the user is signed in but not a subscriber: Labs should render
- the user is signed in and is a subscriber: Labs should **not** render

And that is the current behaviour with the consentless switch on but we (commercial) will have to think about how we handle a full transition to consentless without the switch.

Either way I think webex should not worry about how `isAdFree` is affected by 'reject all' and we will handle ensuring the correct behaviour is followed?

Comment (emma-imber) [reply dotcom-rendering/src/web/layouts/FrontLayout.tsx:352]: Yep completely agree! I'll add this point to the ticket we have around changing the adfree cookie behaviour when we reject all to make sure this is taken into account.
